crates/ruff/src/rules/flake8_type_checking/rules/typing_only_runtime_import.rs

```diff
@@ -342,6 +368,42 @@ pub(crate) fn typing_only_runtime_import(
             }
         };
 
+        if checker.patch(diagnostic.kind.rule()) {
+            diagnostic.try_set_fix(|| {
```

Comment:
Note that we're generating an edit for every diagnostic, so we're generating an edit for every member in an `StmtKind::ImportFrom` statement. We may want to generate one edit for the entire statement, like we do for unused imports... But this does work correctly in practice. The main issue is that if you have:

```py
from foo import bar, bar
```

Then after the fixes, we'll probably give you:

```py
if TYPE_CHECKING:
    from foo import bar
    from foo import baz
```

These will get merged when you run the `isort` rules, so most users will never see this state (or will have it immediately corrected), but I guess it's not ideal (and it does mean that if you're using the LSP, you need to take one Code Action per member).

Reply:
The alternative, as hinted above, is that we generate one edit for the entire import statement. We may also want to then generate one _diagnostic_ for the entire import statement? So this would take some refactoring.